reana_client/utils.py

```diff
@@ -207,31 +207,46 @@ def _validate_serial_workflow_environment(workflow_steps):
     :raises Warning: Warns user if the workflow environment is invalid in serial workflow steps.
     """
     for step in workflow_steps:
-        if ":" in step["environment"]:
-            environment = step["environment"].split(":", 1)
-            environment_image, environment_image_tag = environment[0], environment[-1]
-            if ":" in environment_image_tag:
-                logging.warning(
-                    "Image {} has invalid tag {}".format(
-                        environment_image, environment_image_tag
-                    )
-                )
-                sys.exit(1)
-            elif environment_image_tag in ENVIRONMENT_IMAGE_SUSPECTED_TAGS_VALIDATOR:
-                logging.warning(
-                    "Using '{}' tag is not recommended in {} image.".format(
-                        environment_image_tag, environment_image
-                    )
-                )
-                sys.exit(1)
-        else:
-            logging.warning(
-                "Image {} does not have an explicit tag.".format(step["environment"])
+        _validate_image_tag(step["environment"])
+
+
+def _validate_image_tag(image):
```

Comment:
As far as I understand and according to this [comment](https://github.com/reanahub/reana/issues/248#issuecomment-784004662), all the workflows have image tag in different places. 

Naming can go like this `_validate_serial_image_tag`.

Reply:
This function works regardless of the workflow engine. You pass an image in the format `foo:bar` and it should do the proper validation, not matter if it's serial, Yadage or CWL.

The difference among workflow engines resides in where to find that image string.

Reply:
Okay, Sounds good. Nice approach.

In `yadage` the `image_name` and `image_tag` are not together. To use this function, I will combine them and then pass it to this function 👍

Reply:
Good point, so maybe we can create yet another function that extracts the image name and tag, and depending on the workflow engine it searches for it in one place or another. Then we can pass these two variables to the different validation functions.

Reply:
Since we have `image_name` and `image_tag` inn different places in workflows engines. I am assuming that the new function will also be also be checking the workflow type and then extracting the image name and tag from the dict.

If we will not use them anywhere else extracting them just before we call `_validate_image_tag` would be enough else creating a function that will return image(for example: `x:y`) will be useful.